pkg/async-gateway/main.go

```diff
@@ -20,69 +20,59 @@ import (
 	"flag"
 	"net/http"
 	"os"
-	"strings"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/gorilla/handlers"
 	"github.com/gorilla/mux"
-	"go.uber.org/zap"
-	"go.uber.org/zap/zapcore"
+
+	"github.com/cortexlabs/cortex/pkg/lib/aws"
+	cr "github.com/cortexlabs/cortex/pkg/lib/configreader"
+	"github.com/cortexlabs/cortex/pkg/lib/errors"
+	"github.com/cortexlabs/cortex/pkg/lib/logging"
+	"github.com/cortexlabs/cortex/pkg/lib/telemetry"
+	"github.com/cortexlabs/cortex/pkg/types/clusterconfig"
+	"github.com/cortexlabs/cortex/pkg/types/userconfig"
 )
 
 const (
 	_defaultPort = "8080"
 )
 
-func createLogger() (*zap.Logger, error) {
-	logLevelEnv := os.Getenv("CORTEX_LOG_LEVEL")
-	disableJSONLogging := os.Getenv("CORTEX_DISABLE_JSON_LOGGING")
-
-	var logLevelZap zapcore.Level
-	switch logLevelEnv {
-	case "DEBUG":
-		logLevelZap = zapcore.DebugLevel
-	case "WARNING":
-		logLevelZap = zapcore.WarnLevel
-	case "ERROR":
-		logLevelZap = zapcore.ErrorLevel
-	default:
-		logLevelZap = zapcore.InfoLevel
+var (
+	gatewayLogger = logging.GetLogger()
+)
+
+func Exit(err error, wrapStrs ...string) {
+	for _, str := range wrapStrs {
+		err = errors.Wrap(err, str)
 	}
 
-	encoderConfig := zap.NewProductionEncoderConfig()
-	encoderConfig.MessageKey = "message"
+	if err != nil && !errors.IsNoTelemetry(err) {
+		telemetry.Error(err)
+	}
 
-	encoding := "json"
-	if strings.ToLower(disableJSONLogging) == "true" {
-		encoding = "console"
+	if err != nil && !errors.IsNoPrint(err) {
+		gatewayLogger.Error(err)
 	}
 
-	return zap.Config{
-		Level:            zap.NewAtomicLevelAt(logLevelZap),
-		Encoding:         encoding,
-		EncoderConfig:    encoderConfig,
-		OutputPaths:      []string{"stdout"},
-		ErrorOutputPaths: []string{"stderr"},
-	}.Build()
+	telemetry.Close()
+
+	os.Exit(1)
 }
 
 // usage: ./gateway -bucket <bucket> -region <region> -port <port> -queue queue <apiName>
 func main() {
-	log, err := createLogger()
-	if err != nil {
-		panic(err)
-	}
+	log := logging.GetLogger()
 	defer func() {
 		_ = log.Sync()
 	}()
 
 	var (
-		port        = flag.String("port", _defaultPort, "port on which the gateway server runs on")
-		queueURL    = flag.String("queue", "", "SQS queue URL")
-		region      = flag.String("region", "", "AWS region")
-		bucket      = flag.String("bucket", "", "AWS bucket")
-		clusterName = flag.String("cluster", "", "cluster name")
+		port              = flag.String("port", _defaultPort, "port on which the gateway server runs on")
+		queueURL          = flag.String("queue", "", "SQS queue URL")
+		region            = flag.String("region", "", "AWS region")
+		bucket            = flag.String("bucket", "", "AWS bucket")
+		clusterName       = flag.String("cluster", "", "cluster name")
+		clusterConfigPath = flag.String("cluster-config-path", "", "cluster config path")
```

Comment:
"cluster-config-path" → "cluster-config"

Reply:
this also needs to be added to the K8s spec for the async gateway

Reply:
Sure. Either way sounds good to me. I believe the k8s specs were already modified.